### PR TITLE
Add windows gradle check agent params

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Add windows gradle check agent params

### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/4817
https://github.com/opensearch-project/opensearch-build/issues/2306

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/gradle/gradle-check.jenkinsfile [hunk 1]
```diff
@@ -64,6 +62,24 @@ pipeline {
...
+                    image 'alpine:3'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    currentBuild.description = BUILD_MANIFEST_URL
+                    if (AGENT_LABEL == '') {
+                        currentBuild.result = 'ABORTED'
+                        error("Integration Tests failed to start. Missing parameter: AGENT_LABEL.")
```
Comment (gaiksaya) [inline jenkins/gradle/gradle-check.jenkinsfile:78]: Fix the error message?

File: jenkins/gradle/gradle-check.jenkinsfile [hunk 2]
```diff
@@ -39,6 +29,14 @@ pipeline {
             defaultValue: 'main',
             trim: true
         )
+        // Must use agent with 1 executor or gradle check will show a lot of java-related errors
+        // The c524xlarge is the instance type that has the least amount of errors during gradle check
+        // https://github.com/opensearch-project/OpenSearch/issues/1975
+        choice(
+            name: 'AGENT_LABEL',
+            description: 'Build docker image or not with options.',
```
Comment (gaiksaya) [inline jenkins/gradle/gradle-check.jenkinsfile:37]: Change the description to something relatable.

File: jenkins/gradle/gradle-check.jenkinsfile (hunk 1 above)
Comment (peterzhuamazon) [reply jenkins/gradle/gradle-check.jenkinsfile]: Resolved, thanks.

File: jenkins/gradle/gradle-check.jenkinsfile [hunk 3]
```diff
@@ -39,6 +29,14 @@ pipeline {
             defaultValue: 'main',
             trim: true
         )
+        // Must use agent with 1 executor or gradle check will show a lot of java-related errors
+        // The c524xlarge is the instance type that has the least amount of errors during gradle check
+        // https://github.com/opensearch-project/OpenSearch/issues/1975
+        choice(
+            name: 'AGENT_LABEL',
+            description: 'Choose which jenkins agent to run gradle check',
```
Comment (gaiksaya) [inline jenkins/gradle/gradle-check.jenkinsfile:38]: nit: 'Choose which jenkins agent to run gradle check on'

File: jenkins/gradle/gradle-check.jenkinsfile (hunk 2 above)
Comment (peterzhuamazon) [reply jenkins/gradle/gradle-check.jenkinsfile]: Resolved, thanks.

**Review by gaiksaya — APPROVED** [review]

File: jenkins/gradle/gradle-check.jenkinsfile (hunk 3 above)
Comment (peterzhuamazon) [reply jenkins/gradle/gradle-check.jenkinsfile]: Resolved, thanks.

**Comment by peterzhuamazon** [comment]
Close for now as there are some issues related to the checks.

**Pull request closed without merging**